ibis/pandas/core.py

```diff
@@ -201,7 +225,8 @@ def execute_with_scope(
         aggcontext=aggcontext,
         **kwargs,
     )
-    new_scope = toolz.merge(scope, pre_executed_scope)
+    new_scope = deepcopy(scope)
```

Comment:
Why do we need deepcopy here?

Reply:
This is because both `scope` and `new_scope` are used in the following execution. As suggested, our set method returns nothing but changes the input `scope`, so we will need a deep copy here to create the new_scope, and call `set` to add more stuff in it. In this way, we maintain `scope` unchanged as well.